### PR TITLE
Improved client source documentation

### PR DESCRIPTION
**Pull request by pmeijer**
Far from complete - but added some sought after docs for methods.

- "Moved" `openLibraryOriginInNewWindow` to Library Manager (still available at client).
- Allow passing of empty string to transaction message.

### CODE REVIEW

File: src/docs/client.doc.js [hunk 1]
```diff
@@ -25,4 +27,467 @@
...
+ * @description Remove the library from the current project and branch.
+ * @function removeLibrary
+ * @memberOf Client
+ * @instance
+ * @param {string} libraryName - Name of library to remove.
+ */
+
+/**
+ * @description Rename the library in the current project and branch.
+ * @function removeLibrary
```
Comment (kecso) [inline src/docs/client.doc.js]: renameLibrary

File: src/client/js/client/pluginmanager.js [hunk 2]
```diff
@@ -77,7 +81,7 @@ define([
          * @param {string} pluginId - id of plugin.
```
Comment (kecso) [inline src/client/js/client/pluginmanager.js]: Isn't this description a redundant one now that the doc file also has it?

File: src/client/js/client/pluginmanager.js [hunk 3]
```diff
@@ -107,7 +111,7 @@ define([
...
```
Comment (kecso) [inline src/client/js/client/pluginmanager.js]: Same redundancy question goes for this as well

File: src/docs/client.doc.js (hunk 1 above)
Comment (pmeijer) [inline src/docs/client.doc.js]: Good catch!

File: src/client/js/client/pluginmanager.js (hunk 3 above)
Comment (pmeijer) [inline src/client/js/client/pluginmanager.js]: Let's keep the inlined docs. That file is short as it is and can be useful too.

**Pull request merged**